File: cometa-challenge.frontend/app/services/types/order.ts
export interface OrderResponse {
  created: Date;
  paid: boolean;
  subtotal: number;
  taxes: number;
  discounts: number;
  items: OrderItem[];
  rounds: Round[];
}

interface OrderItem {
  name: string;
  price_per_unit: number;
  total: number;
}

interface Round {
  created: Date;
  items: RoundItem[];
}

interface RoundItem {
  name: string;
  quantity: number;
}

File: cometa-challenge.backend/models/order.py
from pydantic import BaseModel
from typing import List

class OrderItem(BaseModel):
    name: str
    price_per_unit: float
    total: int

class Order(BaseModel):
    rounds: List[dict]
    items: List[OrderItem]
    subtotal: float
    taxes: float
    discounts: float
